src/main/java/data/Task.java

```diff
@@ -76,4 +76,44 @@ public String getStatusIcon() {
     public String getTaskType() {
         return "?";
     }
+
+    /**
+     * Getter for start date for an Event task.
+     * It is a dummy function meant for the Event subclass to override.
+     *
+     * @return Empty string.
+     */
+    public String getStartDate() {
+        return "";
+    }
+
+    /**
+     * Getter for end date for an Event task.
+     * It is a dummy function meant for the Event subclass to override.
+     *
+     * @return Empty string.
+     */
+    public String getEndDate() {
+        return "";
+    }
```

Comment:
These methods can use polymorphism and inherit from a common template, as there are currently several functions that return the same thing